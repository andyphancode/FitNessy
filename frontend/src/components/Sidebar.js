import { Drawer, List, ListItem, ListItemIcon, ListItemText, Box } from '@mui/material';
import HomeIcon from '@mui/icons-material/Home';
import LoginIcon from '@mui/icons-material/Login';
import AppRegistrationIcon from '@mui/icons-material/AppRegistration';
import FitnessCenterIcon from '@mui/icons-material/FitnessCenter';
import LogoutIcon from '@mui/icons-material/Logout';
import { Link } from 'react-router-dom';
import Logo from '../assets/Logo.png';
import { useAuth } from '../context/AuthContext';


const drawerWidth = 240;

function Sidebar() {

  const { currentUser, logout } = useAuth();

  function loggedOut() {
    return (
      <List>
        <ListItem
          key="Home"
          component={Link}
          to={`/`}
          sx={{
            textDecoration: 'none', 
            color: 'inherit',
          }}
        >
          <ListItemIcon>
            <HomeIcon/>
          </ListItemIcon>
          <ListItemText primary={"Home"}/>
        </ListItem>
        {['Login', 'Register'].map((text, index) => (
          <ListItem key={text} component={Link} to={`/${text.toLowerCase()}`}
          sx={{
            textDecoration: 'none', 
            color: 'inherit',
            
          }}
          >
            <ListItemIcon>
              {index === 0 && <LoginIcon />}
              {index === 1 && <AppRegistrationIcon />}
            </ListItemIcon>
            <ListItemText primary={text} />
          </ListItem>
        ))}
      </List>
    );
  };

  function loggedIn() {
    return (
      <List>
        <ListItem
          key="Home"
          component={Link}
          to={`/`}
          sx={{
            textDecoration: 'none', 
            color: 'inherit',
          }}
        >
          <ListItemIcon>
            <HomeIcon/>
          </ListItemIcon>
          <ListItemText primary={"Home"}/>
        </ListItem>
        <ListItem
          key="Workouts"
          component={Link}
          to={`/workouts`}
          sx={{
            textDecoration: 'none', 
            color: 'inherit',
          }}
        >
          <ListItemIcon>
            <FitnessCenterIcon/>
          </ListItemIcon>
          <ListItemText primary={"Workouts"}/>
        </ListItem>    
        <ListItem
          key="Logout"
          sx={{
            textDecoration: 'none',
            color: 'inherit',
            cursor: 'pointer' // Makes it clear this is clickable
          }}
          onClick={logout} // Calls the logout function on click
        >
          <ListItemIcon>
            <LogoutIcon />
          </ListItemIcon>
          <ListItemText primary="Logout" />
        </ListItem>
      </List>
    );
  };

  return (
    <Drawer
      sx={{
        width: drawerWidth,
        flexShrink: 0,
        '& .MuiDrawer-paper': {
          width: drawerWidth,
          boxSizing: 'border-box',
          backgroundColor: '#bab8b4',
        },
      }}
      variant="permanent"
      anchor="left"
    >
        <Box
            sx={{
            display: 'flex',
            alignItems: 'center',
            justifyContent: 'center',
            p: 3
            }}
        >
        <img src={Logo} alt="Logo" style={{ maxWidth: '100%', maxHeight: '150px' }} />
        </Box>

        {currentUser ? loggedIn() : loggedOut()}
    </Drawer>
  );
}

export default Sidebar;