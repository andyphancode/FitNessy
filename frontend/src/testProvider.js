import React from "react";
import UserContext from "./context/UserContext";

const testUser = {
    username: "test",
    email: "test@test.com"
  };

const UserProvider =
  ({ children, currentUser = testUser }) => (
  <UserContext.Provider value={{ currentUser }}>
    {children}
  </UserContext.Provider>
);

export { UserProvider };  