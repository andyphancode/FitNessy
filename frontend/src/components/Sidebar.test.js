import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import Sidebar from "./Sidebar";
import { UserProvider } from "../testProvider";

it("renders without crashing", function () {
    render(
        <MemoryRouter>
            <UserProvider>
                <Sidebar />
            </UserProvider>
        </MemoryRouter>,
    );
});

it("matches snapshot", function () {
    const { asFragment } = render(
        <MemoryRouter>
            <UserProvider>
                <Sidebar />
            </UserProvider>
        </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
});