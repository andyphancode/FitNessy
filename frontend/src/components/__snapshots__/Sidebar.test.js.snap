// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiDrawer-root MuiDrawer-docked css-17pxrdp-MuiDrawer-docked"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft css-12i7wg6-MuiPaper-root-MuiDrawer-paper"
    >
      <div
        class="MuiBox-root css-obxe14"
      >
        <img
          alt="Logo"
          src="Logo.png"
          style="max-width: 100%; max-height: 150px;"
        />
      </div>
      <ul
        class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
      >
        <a
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-wgh4ce-MuiListItem-root"
          href="/home"
        >
          <div
            class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="HomeIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
              />
            </svg>
          </div>
          <div
            class="MuiListItemText-root css-tlelie-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
            >
              Home
            </span>
          </div>
        </a>
        <a
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-wgh4ce-MuiListItem-root"
          href="/workouts"
        >
          <div
            class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="FitnessCenterIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20.57 14.86 22 13.43 20.57 12 17 15.57 8.43 7 12 3.43 10.57 2 9.14 3.43 7.71 2 5.57 4.14 4.14 2.71 2.71 4.14l1.43 1.43L2 7.71l1.43 1.43L2 10.57 3.43 12 7 8.43 15.57 17 12 20.57 13.43 22l1.43-1.43L16.29 22l2.14-2.14 1.43 1.43 1.43-1.43-1.43-1.43L22 16.29z"
              />
            </svg>
          </div>
          <div
            class="MuiListItemText-root css-tlelie-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
            >
              Workouts
            </span>
          </div>
        </a>
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-iinlml-MuiListItem-root"
        >
          <div
            class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="LogoutIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="m17 7-1.41 1.41L18.17 11H8v2h10.17l-2.58 2.58L17 17l5-5zM4 5h8V3H4c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h8v-2H4z"
              />
            </svg>
          </div>
          <div
            class="MuiListItemText-root css-tlelie-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
            >
              Logout
            </span>
          </div>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
